src/metadata/trackmetadatataglib.cpp

```diff
@@ -1076,79 +1084,84 @@ Result writeTrackMetadataIntoFile(const TrackMetadata& trackMetadata, QString fi
     qDebug() << "Writing track metadata into file" << fileName << "of type" << fileType;
 
     QScopedPointer<TagLib::File> pFile;
+    bool tagWritten = false;
 
     if (kFileTypeMP3 == fileType) {
         QScopedPointer<TagLib::MPEG::File> pMPEGFile(
                 new TagLib::MPEG::File(fileName.toLocal8Bit().constData()));
         bool defaultID3V2 = true;
         if (hasAPETag(*pMPEGFile)) {
-            writeTrackMetadataIntoAPETag(pMPEGFile->APETag(), trackMetadata);
+            tagWritten |= writeTrackMetadataIntoAPETag(pMPEGFile->APETag(), trackMetadata);
```

Comment:
This looks like you care about individual bits. On the other hand it is a nice on-liner :-/
Do you actually need to care about the previous state? From a short skim over the code it looks like we are only pass such statement once.